core/arch/x86/mangle.c

```diff
@@ -2565,18 +2570,21 @@ mangle_float_pc(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
         if (op == OP_fnsave || op == OP_fnstenv) {
             opnd_set_disp(&memop, opnd_get_disp(memop) + FNSAVE_PC_OFFS);
             opnd_set_size(&memop, OPSZ_4);
+            /* XXX i#3307: necessary to instr_set_translation_mangling_epilogue? */
```

Comment:
I think it's ok to re-execute these state save instructions.  Certainly it would be wrong to xl8 to the next instr w/o having fixed up the stored PC.

Reply:
Comment removed.